### PR TITLE
Add module for the dhall config language

### PR DESCRIPTION
**Pull request by rgrinberg**
Title explains everything

### CODE REVIEW

File: modules/lang/dhall/config.el [hunk 1]
```diff
@@ -0,0 +1,11 @@
+;;; lang/dhall/config.el -*- lexical-binding: t; -*-
+
+(use-package! dhall-mode
```
Comment (hlissner) [inline modules/lang/dhall/config.el:3]: Add `:defer t`, otherwise the package is eagerly loaded at startup.

File: init.example.el [hunk 2]
```diff
@@ -139,6 +139,7 @@
        ;;kotlin            ; a better, slicker Java(Script)
        ;;latex             ; writing papers in Emacs has never been so fun
        ;;lean
+       ;;dhall
```
Comment (hlissner) [inline init.example.el:142]: Please sort this alphabetically (below `dart`)

File: modules/lang/dhall/config.el [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  :config
+  (set-repl-handler! 'dhall-mode #'dhall-repl-show)
+  (map! :after dhall-mode
```
Comment (hlissner) [inline modules/lang/dhall/config.el:6]: Remove `:after dhall-mode`, it's redundant with nesting this `map!` in `dhall-mode`'s `:config` block.

Comment (hlissner) [inline modules/lang/dhall/config.el:6]: Please add `(setq dhall-format-at-save (featurep! :editor format +onsave))` on the line between `set-repl-handler!` and `map!`.

**Review by hlissner — CHANGES_REQUESTED** [review]
Also:

1. Please add an entry for `:lang dhall` to `docs/modules.org`.
2. And add a dummy README.org using the `__doom-readme` snippet for this module.

**Comment by rgrinberg** [comment]
Thanks for the review!

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Sorry for the _tremendously_ late reply. Thanks for the PR!

**Pull request merged**